### DISCUSSION TITLE
Impedance matching newbie help: in general, and to Wilkinson power divider specifically

### DISCUSSION THREAD

**Discussion by oberstet** [thliebig/openEMS-Project#178] [General]
Hi all,

I'm a RF newbie that got into RF for reasons, and after some weeks of getting into this stuff in general, and openEMS specifically, I now begin to see a light;)

One area of the RF antenna/device I am designing is GNSS, and for this part, I now made progress but am again at a point where I am pretty much lost.

Any helps would be highly appreciated! Helping a RF newbie will improve your Karma, promise! ;) Seriously, I could offer deep know how in various software fields in exchange!

Anyways, details below, thank you for your attention, time and any help!

Cheers,
/Tobias

Current status is:

I have openEMS models for a multi-band GNSS antenna which looks good to me:

![gnss-dipole-1575](https://github.com/thliebig/openEMS-Project/assets/233340/8698bb9d-a09c-4899-a682-8ef9319f014f)

![gnss-dipole-1227](https://github.com/thliebig/openEMS-Project/assets/233340/96807a1e-5c25-4466-92be-bf7e72ce707f)

and I do have a design for the complete active antenna

[rfminer-gnss-questions1.pdf](https://github.com/thliebig/openEMS-Project/files/13941909/rfminer-gnss-questions1.pdf)

*That is, I do have the parts marked "green" in above, but I am lost in the part marked "red" ...*

Does that look good on a block diagram level? How do I design that?  I mean, I need these parts and values.

Please understand: I do have _some_ clue about electronics, and I am a quick learner, but I have pretty much zero RF experience, and usually live in my editor all day and press RETURN from time to time;)

**Comment by VolkerMuehlhaus** [comment]
Tobias, I am confused by your design. Why do you want to use the coupler, if there are two separate bands with one dedicated antenna and dedicated SAW per band? Using the coupler you will loose 3dB signal, half power from each antenna goes into the "wrong" path where it is rejected by the SAW anyway. From my understanding, the coupler between antennas and SAW should be removed, and each antenna should feed one path directly.

A coupler is indeed useful to combine the outputs of the two signals paths, after the LNAs.

Second question: Why did you design the antenna for non-standard impedance? It is common practice to design the RF interfaces for 50 Ohm, which is also used in measurement equipment. Of course, your antenna impedance can be matched to 50 Ohm using LC matching network, it is not too far off. But in general, LC matching networks cover limited bandwith only, and should be included for a realistic bandwith estimate.

This forum might be the wrong place for such generic RF topics not related to openEMS. You might want to have a look at the RF section at edaboard. It is sort of lazy student forum, many students look there for support to get their RF design homework done, but there are several RF pros around as well.

Cheers,
Volker

**Reply by oberstet** [reply]
Thank you so much (again) for your guidance Volker! This helps me a lot, following the right path;)

>  I am confused by your design ... each antenna should feed one path directly

My current design, with this coupler etc, is mimicked after how a specific commercially available active GNSS antenna is designed. So it is not sth I "invented";)

*I want to receive GNSS on 2 bands: L1 (~1.2GHz) and L5 (~1.5GHz), and I want to do so _concurrently_.*

In such a situation, what seems to be commonly done is: have 2 antenna elements tuned to those 2 bands.

I do have those antenna elements now (in openEMS). Step 1 done.

The problem is: commercially available GNSS _modules_ (which I wanted to use ... because "simpler" compared to using bare bones GNSS chips directly ... I thought) however _only have one RF input, not two_.

This is probably the reason most (all?) GNSS active antenna designs I looked at have _one_ RF out with this fancy coupler thing internally, which combines the 2 RFs from the 2 antenna elements, into one RF first, which only then is fed into the downstream GNSS module (which has 1 RF input).

I'd love to simply it if possible! Less stuff means less things I need to figure out. Moving complexity to _software_ is totally cool!

I have now looked deeper into this stuff (GNSS modules internal structure), and as it turns out: those modules have in fact 2 RF inputs in their GNSS chips *internally*!

pls see page 2/3 I added here:

[rfminer-gnss-questions1.pdf](https://github.com/thliebig/openEMS-Project/files/13960345/rfminer-gnss-questions1.pdf)

----

So what I am now figuring out is: what are the consequences of using a bare bones GNSS chip with 2 separate RF inputs for the 2 bands (like the Synaptics SYN4778 dual-frequency (L1/L5) GNSS receiver).

If I follow this path, I don't need a coupler or what, and I should be able to connect the 2 antenna elements separately to the GNSS chip. 

Possibly only with one L1 SAW+LNA and one L5 SAW+LNA. Not sure.

Ideally, I would want to directly connect my 2 antenna elements to 1 chip via 2 RF paths (50 Ohm transmission lines). basically like you seem to hint at with

> the coupler between antennas and SAW should be removed, and each antenna should feed one path directly.

-----

> Second question: Why did you design the antenna for non-standard impedance?

Up to now, I have only designed antenna elements *for a desired resonance frequency and bandwidth*.

This works for me, I have enough clues and know my way with openEMS. Good.

I have _not_ yet looked at trying to modify the antenna to hit a desired impedance.

The impedance for me is just "a result" .. like the S11. And the impedances resulting are within ranges like the EM textbooks say (for dipole and monopoles etc)

Obviously, what you hint at goes further: not only optimize for resonance frequency + bandwidth, but at the same time for desired target impedance, right?

If so, I get that, makes sense.

Given my current state of understanding, I would note: I should be able to optimize for 50 Ohm _at the resonance frequency_, but _not_ over the whole bandwidth of the antenna (S11 < -10db).

I could verify the impedance range of standard antennas over bandwidth as written in textbooks. It can range from broadly 30 to 100 Ohm.

*Is it even possible to design _one_ antenna element resonant at `fc` with bandwidth `f0` to `f1`, _and_ impedance 50 Ohm, not only at `fc`, but for every frequency within `f0` to `f1`?*

-----

Now, how do I modify my antenna so that it has 50 Ohm at `fc`? As next step.

Currently it has ~62 Ohm.

*So does that mean I need a series capacitance to hit 50 Ohm?*

I will try that. "Seems" straight forward. I will see;)


-----

> You might want to have a look at the RF section at edaboard.

fantastic!!!

https://www.edaboard.com/forums/rf-microwave-antennas-and-optics.26/

btw: sorry for spamming the forum here, not intended, I will move to edaboard.

fwiw: I am not lazy, just lost / a newbie;)

**Reply by VolkerMuehlhaus** [reply]
> This is probably the reason most (all?) GNSS active antenna designs I looked at have one RF out with this fancy coupler thing internally, which combines the 2 RFs from the 2 antenna elements, into one RF first, which only then is fed into the downstream GNSS module 

The difference to your design is that the "fancy coupler thing" seems to be a diplexer ("Frequenzweiche"), which can split the signals with much lower loss. From the NEO-F10N integration manual: "The integrated RF circuit includes an L1, L5 dual-band **diplexer SAW filter** followed by a low-noise amplifier (LNA) and a second SAW filter stage at both GNSS bands". 

The diplexer directs all L1 signal to the L1 path, and all L5 signal to the L5 path. Your power splitter directs half power to each channel, regardless of frequency => 3dB loss

Signal to noise ratio is critical in GNSS, and your antenna already has 3dB loss due to linear polarization instead of circular polarization. It is possible to use linear antennas if your overall signal budget is ok, but 3dB loss here and another 3dB loss there will add up ...

> Is it even possible to design one antenna element resonant at fc with bandwidth f0 to f1, and impedance 50 Ohm, not only at fc, but for every frequency within f0 to f1?

This is the target, and we try to get close, but you will not have 50 ohm exactly across the entire band. We usually circle around the 50 Ohm point in Smith chart, at acceptable distance within the entire band. Note the environment (like PCB ground plane) will also change impedance levels. 

> So does that mean I need a series capacitance to hit 50 Ohm?

60 Ohm is still acceptable, although not on target. I just wanted to mention that phantastic matching (S11) to an **arbitrary** impedance means nothing, we want to match for a given target value.

To do the impedance matching, you would need L and C, online calculator [here](https://www.analog.com/en/design-center/interactive-design-tools/rf-impedance-matching-calculator.html)

> I am not lazy, just lost

Sorry for the misunderstanding: the "lazy student forum" really referred to that forum, not to you! There are plenty of students there who keep posting their homework stuff, and they really should know better because they hopefully attended the seminars.

**Reply by oberstet** [reply]
> Signal to noise ratio is critical in GNSS, ...

yeah, I really should take that exactly as you say, and seriously.

I understand that my combination of "power splitter" + "diplexer" isn't doing good.

fwiw, I want to do it "right", even if it means more head scratching. thus, I want to follow what you hinted at: have 2 antenna elements, and separate paths directly to the GNSS chip on separate (L1 vs L5) RF inputs.

I have found (my impression) some suitable SAW filters made by Qorvo .. they are expensive .. but I have to make compromises. anyways:

![grafik](https://github.com/thliebig/openEMS-Project/assets/233340/ff9ec0fd-a435-471a-8037-6bea1af08212)

----

I am now trying to find suitable LNAs, and I'm also wondering, should it be

1. antenna element (tuned to 50 Ohm) => LNA => SAW => PCB transmission line => GNSS chip, *or should it be*
2.  antenna element (tuned to 50 Ohm) => SAW => LNA => PCB transmission line  => GNSS chip

And this presumes already that my general idea of placing LNA+SAW closely to the antenna element feed point, have a cm's long transmission line then, leading to the GNSS chip in a separated PCB area makes sense ...

----

> I just wanted to mention that phantastic matching (S11) to an arbitrary impedance means nothing, we want to match for a given target value.

I get it. Now. Thanks for the strong hint / direction!

I need to weave that in .. up till now I only considered S11, resonance freq and bandwidth, and I can control/design for those .. but impedance I need to take in yet. I feel like I am close;) maybe a chimera, maybe I again underestimate things, .. will see.

> Sorry for the misunderstanding ..

no worries, I understand. it's just I am paranoid, and I myself absolutely hate people coming around not doing proper homework even though they are beginners. if they do their homework, _then_ I am willing to help (I am talking about software things where I know a ton). so I want to meet that level myself and not leave a wrong impression;)

**Comment by oberstet** [comment]
alright, I've taken in impedance (of 50 ohm) as an optimization goal, and managed to hit 53 ohm (and with driving impedance of 50 ohm for the lumped port):

![grafik](https://github.com/thliebig/openEMS-Project/assets/233340/4c01409d-3fcf-4e4a-ad3d-3bc099d65088)

but not yet 50 ohm, and also loosing a some dB's , but hey, this is WIP

----

moreover: I now understand that I also need to match the balanced output of the dipole to the unbalanced input of the LNA .. yet another thing I kinda ignored / was unaware of;)

anyways: rather than to continue optimizing to exactly 50 ohm I will address both balanced-to-unbalanced and 50 ohm matching by designing what seems to be called a "microstrip balun" ...

thanks again Volker for steering me in the right direction! closing this discussion, if so, then I'd ask on the forum you've hinted at.

**Reply by VolkerMuehlhaus** [reply]
Hi Tobias,

a microstrip line balun would take quite a lot of PCB area, so I would recommend to create it from SMD elements to save PCB space. 
https://www.changpuak.ch/electronics/LumpedBalunCalculator.php

Have a great weekend!
Volker

**Comment by oberstet** [comment]
once again, thank you so much!! this is invaluable. I will add that resource to my docs and consider going down that path as well!

> thing is, because of this other (non-gnss, UHF) antenna element I also need, I kinda redesigned/rescheduled my actual first goals, and "size" isn't a main factor for now anymore. I will use a dual LPDA, and the "dual" includes "phase synchronized" all the way up. because I forgot: yes, I need the bandwidth (I now design for 400-2800 mhz), but I also need it as a 2 element phased array;) anyways, I am getting there - even if much more slowly than I anticipated. it is definitely a tricky field;)

**Comment by oberstet** [comment]
> now that I started this slightly OT discussion here, I'll conclude / stop commenting / will move to that other forum next time, promise

actually, your "go for lumped balun" hint and the calculator you linked are fantastic!

not only for size (which isn't prio 1 for me anymore atm), but even more so: simplification! digging into microstrip baluns is yet another topic I did not wait for;) I can use my gnss dipoles exactly as they are, with off-50 ohm impedance!

seems like components are readily available:

![Bildschirmfoto vom 2024-01-19 15-26-41](https://github.com/thliebig/openEMS-Project/assets/233340/e7ed706b-aebb-47ad-8ee5-fce9aceedb4a)
